### PR TITLE
plugin.audio.calmradio - 0.1.4

### PR DESCRIPTION
**Pull request by nire0510**
New add-on for calmradio.com internet radio station.

### CODE REVIEW

**Comment by phil65** [comment]
General structure looks fine now.
One small "warning": xbmcswift2 wasnt maintained for quite some time if I am not mistaken, so perhaps it´s not the best idea to use that framework for plugin-development unless someone updates it.

**Comment by nire0510** [comment]
@phil65 thank you!
Actually i've been using xbmcswift2 for quite long, have dozens of add-ons I developed with which work fine on Isengard (vanilla & openelec) & Jarvis.

**Comment by phil65** [comment]
I somewhere read that it didnt catch up with some of our core additions (like using setArt() for artwork).
Anyways, not a blocker, just a small heads-up, perhaps you want to be the one who updates that framework :)

**Comment by nire0510** [comment]
@phil65 sounds like an idea! 💡 
I'll dive in and see if it's feasible to take ownership on that project.
Thank you for your help with the PR submission!

File: plugin.audio.calmradio/resources/language/English/strings.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="utf-8" standalone="yes"?>
```
Comment (phil65) [inline plugin.audio.calmradio/resources/language/English/strings.xml]: xml language format is deprecated, please convert this to a .po file.

File: plugin.audio.calmradio/user.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        return False
+
+    def invalidate(self):
+        """
+        Logs current user out
+        :return:
+        """
+        del self.auth['timestamp']
+        del self.auth['token']
+        self.auth.sync()
```
Comment (phil65) [inline plugin.audio.calmradio/user.py]: some files are missing a newline at EOF, and please convert strings.xml to strings.po.
Apart from that, it looks fine to me.

Comment (nire0510) [inline plugin.audio.calmradio/user.py]: Can you send me an example of .po file?
Do i need to submit a new PR after these changes?

Comment (phil65) [inline plugin.audio.calmradio/user.py]: You can use this: https://github.com/alanwww1/xbmc-xml2po

You dont need to open a new PR, just overwrite your "plugin.audio.calmradio" branch by force-pushing, the PR will get updated automatically then.

Comment (phil65) [inline plugin.audio.calmradio/user.py]: Compiled windows version: https://github.com/alanwww1/xbmc-xml2po/downloads

**Comment by learningit** [comment]
As you are calling this the "official" add-on for calmradio, can you explain your relationship with them?

**Comment by nire0510** [comment]
@learningit sure, I suggested my (free) services just to be able to enjoy listening to Calm Radio on Kodi. Gladly the CEO was thrilled about the option to offer his services through another popular platform.

**Comment by nire0510** [comment]
@phil65 @learningit hello guys, I made all the changes you asked, I have even rewritten the project and now it is no longer based on xbmcswift! strings.xml replaced with strings.po and added those blank lines at the end of every *.py file. Will you kindly approve my PR now?

Thanks!

File: plugin.audio.calmradio/addon.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    </extension>
+    <extension point="xbmc.addon.metadata">
+        <platform>all</platform>
+        <language>en</language>
+        <summary>Calm Radio Multimix</summary>
+        <description>Listen to to over 140 incredible channels including Atmosphere channels, specially designed for sleep, concentration and for ringing-in-the-ear sufferers</description>
+        <disclaimer lang="en">This is an official add-on of calmradio.com</disclaimer>
+        <license>GNU GENERAL PUBLIC LICENSE Version 2, June 1991</license>
+        <website>http://calmradio.com/</website>
+        <email>support@calmradio.com</email>
```
Comment (phil65) [inline plugin.audio.calmradio/addon.xml]: please add a `<source>` entry.

**Comment by phil65** [comment]
...and please add a solid background to your icon.png.  After that it's good to go I think. :+1:

**Comment by nire0510** [comment]
@phil65 done!

**Comment by phil65** [comment]
Thx. From my side all looks fine now. @learningit, mind takin another last look before pressing the button?

**Comment by nire0510** [comment]
@phil65 @learningit Hi guys, are there any issues left which still prevent this add-on from being merged?
Thanks!

**Comment by phil65** [comment]
Be patient. :) From my side all is fine, @learningit 	will press the button after a final check from him.

**Comment by nire0510** [comment]
Sure :) thanks for updating!

**Comment by learningit** [comment]
I'm ok with this for now. I would ask that you try to see if you could use the images off the website for the next update. This would speed up updates from the repo (smaller zip to load) and would keep the add-on in synch with the website. I see that there would be a mapping involved, but you may be able to get the website to provide that for you - particularly if this is an "official" add-on.

**Comment by nire0510** [comment]
Hi @learningit - thanks for merging my PR, really excited to give something back to Kodi community :)
As for the images, it is an official add-on, but the thing is that the blurred images are not part of the website (i created them based on the thumbnails the API provides) and hence are not available via URL.
I will try to think about some other alternative, maybe to host these files on my hosting.
Thanks again, @learningit @phil65 for the patience and guidance!

I would appreciate it if you could write in few words when will the add-on be available for all users and if version PRs work the same.

Nir

**Comment by phil65** [comment]
Should be available within 24h.
Update PRs work the same, yes.

**Pull request merged**